app/code/Magento/SalesGraphQl/Model/Resolver/Invoices.php

```diff
@@ -41,10 +41,31 @@ public function resolve(
             $invoices[] = [
                 'id' => base64_encode($invoice->getEntityId()),
                 'number' => $invoice['increment_id'],
+                'comments' => $this->getInvoiceComments($invoice),
```

Comment:
Actually invoices should not expose comments
They are not in luma and we should deprecate comments in invoice graphql and don't do this fix

Reply:
@cpartica are you sure they're not present in Luma?
https://github.com/magento/magento2/blob/5f0516612488bda34d4c0bf2aa863a3510c374b0/app/code/Magento/Sales/view/frontend/layout/sales_order_invoice.xml#L32
https://github.com/magento/magento2/blob/5f0516612488bda34d4c0bf2aa863a3510c374b0/app/code/Magento/Sales/view/frontend/templates/order/invoice/items.phtml#L51

Reply:
Also:
https://github.com/magento/magento2/blob/5f0516612488bda34d4c0bf2aa863a3510c374b0/app/code/Magento/Sales/Block/Order/Invoice/Items.php#L98-L107

Reply:
@cpartica I have confirmed that invoice comments are indeed visible in Luma - please see screenshot below taken from a fresh 2.4-develop instance.
![image](https://user-images.githubusercontent.com/40189797/118901353-35713300-b952-11eb-9a10-e46f73154522.png)

Comments missing from the GraphQL response is clearly a bug, and I'm not sure why it's taking so long for this very simple PR to be merged - it's been over 4 months now since it was approved.

Reply:
Hi @cpartica, indeed the invoice comments are shown on Luma. Probably you missed the following checkbox, which makes the comment visible on the storefront.
 
![image](https://user-images.githubusercontent.com/15868188/120145812-ccf54200-c1ec-11eb-81c6-69bd06685b95.png)

Reply:
Hey @cpartica! Looks like @aligent-lturner is right and invoice comments are available to view on the storefront (magento has a visibility modifier for the comments).